### DISCUSSION TITLE
I can't figure out how to properly handle with nuxt server errors

### DISCUSSION THREAD

**Discussion by rrAlfaro** [nuxt/nuxt#28857] [Questions]
Hi! 

I'm studying about nuxt server structure and Im kind new to it, so I would be glad if you can point out what am I doing wrong or if I am just lost at it.

Preveously, I was doing all my requests directly into the page script like this: 
```
<script setup>
  const config = useRuntimeConfig();
  const token = useCookie('token')
  const route = useRoute()
 
  const { data: authors } = await useFetch(`${config.public.APIURL}/api/admin/authors/${route.params.id}`, {
    credentials: 'include',
    headers: {
      Accept: 'application/json',
      Authorization: `Bearer ${token.value}`
    },
    method: 'GET',
  })
</script>
```

Then I felt the need to organize it, as my system was growing. So i dived into nuxt server folder as it seems to me to be the right way. (Tell me if I'm wrong)

Ok then, in my server folder I created api/ authors/ index.vue

Inside index.vue I throwed my request code: 
```
export default defineEventHandler(async (event) => {
  const config = useRuntimeConfig();
  const token = getCookie(event, "token");

  const response = await $fetch(`${config.public.APIURL}/api/admin/authors`, {
    credentials: "include",
    headers: {
      Authorization: `Bearer ${token}`,
    },
  });

  const idealResponse = response.map((item) => {
    return {
      id: item.id,
      name: item.name,
      image_path: item.image_path,
      created_at: item.created_at,
    };
  });

  return idealResponse;
});
```

Then in my componente I called it like this: 
`const { data: authors, error } = await useFetch('/api/authors')`

First I would like to ask if this is the proper way to do it.
Second, my biggest issue is with error handling.
For example, if I do the request call directly in my component, as I used to do, I got the proper errors for the problems, like, if my token was expired, 401 unauthorized was throw. BUT, when with this server structure, I always get status 500 regardless of the error.

I get it that doing this is like a proxy, cause in network the request is being send to localhost, so I tried to do a try catch to see if I could get the right error and then send it to my component, but everywhere was throwing status 500 if something is not right, like token, or wrong route, etc.

Could you guys help me understand this? 
Thank you! 

Sorry for bad english too

**Comment by oneminch** [comment]
If your server file is `api/authors/index.vue`, that could be the issue. Server files are either javascript or typescript files, so your file must be `index.js` or `index.ts`.
